wgpu-types/src/lib.rs

```diff
@@ -3171,12 +3172,28 @@ impl DepthBiasState {
     }
 }
 
+impl Hash for DepthBiasState {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.constant.hash(state);
+        self.slope_scale.to_bits().hash(state);
+        self.clamp.to_bits().hash(state);
+    }
+}
+
+impl PartialEq for DepthBiasState {
+    fn eq(&self, other: &Self) -> bool {
+        (self.constant == other.constant)
+            & (self.slope_scale == other.slope_scale)
```

Comment:
This is a subtle one: if we use to_bits() for the hash implementation, we also need to use it for the eq implementation. One of the rules of `Hash` is that if `a == b` then `hash(a) == hash(b)` and for negative zero and other such values that is broken.

Reply:
I see, so in this case what should we do? use to_bits() assuming the value never reaches zero and below or handle the problem case manually ?

Reply:
We should just compare the bits as is and not worry about the weird extra cases.